### PR TITLE
Fix order references

### PR DESCRIPTION
**Pull request by elythyr**
Order the result of the find references feature.
Fix https://github.com/phpactor/phpactor/issues/1105

Update both the RPC handler and the `phpactor#quickfix#fzf` strategy, see the related commit for an explanation but the problem was that the current implementation did not preserve the order of the provided entries.

### CODE REVIEW

**Comment by elythyr** [comment]
Any news on the CI ?
I guess you're waiting to fix it before merging anything else ?

File: lib/Extension/ClassMover/Rpc/ReferencesHandler.php [hunk 1]
```diff
@@ -268,19 +268,46 @@ private function performFindOrReplaceReferences(
...
             case Symbol::METHOD:
-                return $this->memberReferences($filesystem, $symbolContext, ClassMemberQuery::TYPE_METHOD, $source, $replacement);
+                [$source, $references] = $this->memberReferences($filesystem, $symbolContext, ClassMemberQuery::TYPE_METHOD, $source, $replacement);
+                break;
             case Symbol::PROPERTY:
-                return $this->memberReferences($filesystem, $symbolContext, ClassMemberQuery::TYPE_PROPERTY, $source, $replacement);
+                [$source, $references] = $this->memberReferences($filesystem, $symbolContext, ClassMemberQuery::TYPE_PROPERTY, $source, $replacement);
+                break;
             case Symbol::CONSTANT:
-                return $this->memberReferences($filesystem, $symbolContext, ClassMemberQuery::TYPE_CONSTANT, $source, $replacement);
```
Comment (dantleech) [inline lib/Extension/ClassMover/Rpc/ReferencesHandler.php:277]: could rename this method to f.e. `doPerformFindOrReplace` and decorate it with the sorting functionality.

File: lib/Extension/ClassMover/Rpc/ReferencesHandler.php [hunk 2]
```diff
@@ -268,19 +268,46 @@ private function performFindOrReplaceReferences(
...
-            'Cannot find references for symbol type "%s"',
-            $symbolContext->symbol()->symbolType()
-        ));
+        return [$source, $this->sortReferences($references)];
+    }
+
+    public function sortReferences(array $fileReferences): array
+    {
+        // Sort the references for each file
+        \array_walk($fileReferences, function (array &$fileReference) {
```
Comment (dantleech) [inline lib/Extension/ClassMover/Rpc/ReferencesHandler.php:295]: no need to use an FQN here?

**Comment by dantleech** [comment]
CI should work again now - had to migrate from travis-ci.org to travis-ci.com

**Comment by dantleech** [comment]
btw I'd sooner replace this functionality with the new [reference finder as used by the languaeg server](https://github.com/phpactor/language-server-phpactor-extensions/blob/d5c6c4d3d5f3e8683ea30a8908c348034517de7d/lib/LanguageServerReferenceFinder/Handler/ReferencesHandler.php#L26) finally - ideally if the abstraction could also fall back to the non-indexed strategy here. But first it should also support search+replace.

File: lib/Extension/ClassMover/Rpc/ReferencesHandler.php (hunk 2 above)
Comment (elythyr) [reply lib/Extension/ClassMover/Rpc/ReferencesHandler.php:295]: All the references reference the same name so I'm not sure on which FQN we could sort.
The only things we are provided with are the file path and the positions.
So we sort the references in each files by position and then all files by alphabetical order.

File: lib/Extension/ClassMover/Rpc/ReferencesHandler.php (hunk 1 above)
Comment (elythyr) [reply lib/Extension/ClassMover/Rpc/ReferencesHandler.php]: Done, if I understood what you meant ;)

**Comment by elythyr** [comment]
> btw I'd sooner replace this functionality with the new reference finder as used by the languaeg server finally - ideally if the abstraction could also fall back to the non-indexed strategy here. But first it should also support search+replace.
There is already a PR for the LSP extension: https://github.com/phpactor/language-server-phpactor-extensions/pull/5

So switching will not change anything regarding the sorting.

**Comment by elythyr** [comment]
I can't figure out why the CI tests don't pass.
I removed my local packages, clean my vendors, install the packages from the lockfile, symlink the vendor and the lock file to the indexer-extension (as done in the `testdeps` script) and all tests passes...

File: lib/Extension/ClassMover/Rpc/ReferencesHandler.php (hunk 2 above)
Comment (dantleech) [reply lib/Extension/ClassMover/Rpc/ReferencesHandler.php:305]: i mean the `array_walk` doesn't need to be absolutely referenced?

Comment (dantleech) [reply lib/Extension/ClassMover/Rpc/ReferencesHandler.php:305]: also should be private?

Comment (elythyr) [reply lib/Extension/ClassMover/Rpc/ReferencesHandler.php:305]: ho yes, bad habit from work with the `\` and public well... It seems I'm out of my game :)

**Comment by elythyr** [comment]
Up here too, to have the same behavior with LSP and RPC

**Pull request merged**